### PR TITLE
Improve IDE compatibility

### PR DESCRIPTION
**Pull request by dereuromark**
Less yellow in the IDE, more typehinting now included :)

### CODE REVIEW

File: src/Controller/Component/AuthComponent.php [hunk 1]
```diff
@@ -877,7 +877,7 @@ public function constructAuthenticate()
      *
      * @param \Cake\Auth\Storage\StorageInterface|null $storage Sets provided
      *   object as storage or if null returns configured storage object.
-     * @return \Cake\Auth\Storage\StorageInterface|null
+     * @return \Cake\Auth\Storage\StorageInterface|\Cake\Core\InstanceConfigTrait|null
```
Comment (dereuromark) [inline src/Controller/Component/AuthComponent.php]: Unfortunately, this is needed for getConfig() and setConfig calls on this storage() to be typehinted and found as valid.

File: src/Controller/Component/SecurityComponent.php [hunk 2]
```diff
@@ -510,12 +511,12 @@ protected function _debugPostTokenNotMatching(Controller $controller, $hashParts
             'Missing field \'%s\' in POST data'
         );
         $expectedUnlockedFields = Hash::get($expectedParts, 2);
-        $dataUnlockedFields = Hash::get($hashParts, 2) ?: [];
+        $dataUnlockedFields = Hash::get($hashParts, 2) ?: null;
```
Comment (markstory) [inline src/Controller/Component/SecurityComponent.php]: Why are you defaulting to `null` and then casting?

Comment (dereuromark) [reply src/Controller/Component/SecurityComponent.php]: Warning about string Operation on array in next line. Switching types on the same var here.

Comment (markstory) [reply src/Controller/Component/SecurityComponent.php]: Ok. I misread explode as implode derp.

**Pull request merged**